### PR TITLE
feat:state:Ignore market balance after nv23

### LINKED ISSUE #11921: Implement FIP 0065
### Checklist

- [X] This is **not** brainstorming ideas. If you have an idea you'd like to discuss, please open a new discussion on [the lotus forum](https://github.com/filecoin-project/lotus/discussions/categories/ideas) and select the category as `Ideas`.
- [X] I **have** a specific, actionable, and well motivated feature request to propose.

### Lotus component

- [X] lotus daemon - chain sync
- [ ] lotus fvm/fevm - Lotus FVM and FEVM interactions
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt/WinningPoSt)
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### What is the motivation behind this feature request? Is your feature request related to a problem? Please describe.

FIP 0065 ignoring f05's contribution to locked FIL is accepted.

### Describe the solution you'd like

Implement the FIP

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ZenGround0**
## Related Issues

Closes #11921 
## Proposed Changes

Drop builtin market calculation from locked balance calculation after nv23

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [x] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: chain/stmgr/supply.go [hunk 1]
```diff
@@ -303,11 +303,14 @@ func getFilPowerLocked(ctx context.Context, st *state.StateTree) (abi.TokenAmoun
...
 }
 
-func GetFilLocked(ctx context.Context, st *state.StateTree) (abi.TokenAmount, error) {
-
-	filMarketLocked, err := getFilMarketLocked(ctx, st)
-	if err != nil {
-		return big.Zero(), xerrors.Errorf("failed to get filMarketLocked: %w", err)
+func GetFilLocked(ctx context.Context, st *state.StateTree, height abi.ChainEpoch) (abi.TokenAmount, error) {
+	filMarketLocked := big.Zero()
+	if height <= build.UpgradeAussieHeight {
```
Comment (rvagg) [inline chain/stmgr/supply.go:308]: how about we switch this around with a view to the future, where the common case is `height > build.UpgradeAussieHeight` and comes first, where we just do a simple one-line `return getFilPowerLocked(ctx, st)`, then we don't need `filMarketLocked=0`.

File: chain/stmgr/supply.go [hunk 2]
```diff
@@ -303,11 +303,14 @@ func getFilPowerLocked(ctx context.Context, st *state.StateTree) (abi.TokenAmoun
 	return pst.TotalLocked()
 }
 
-func GetFilLocked(ctx context.Context, st *state.StateTree) (abi.TokenAmount, error) {
-
-	filMarketLocked, err := getFilMarketLocked(ctx, st)
-	if err != nil {
-		return big.Zero(), xerrors.Errorf("failed to get filMarketLocked: %w", err)
+func GetFilLocked(ctx context.Context, st *state.StateTree, height abi.ChainEpoch) (abi.TokenAmount, error) {
```
Comment (arajasek) [inline chain/stmgr/supply.go:306]: Please refactor this to receive a network version instead, and make the change based on that. I know we have a lot of height comparisons in this area of the code, but that's quite bad and we shouldn't be adding any more.

The `StateManager` can lookup network versions based on the provided height and supply it here.

**Review by arajasek — CHANGES_REQUESTED** [review]

File: itests/supply_test.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	require.NoError(t, err)
+	ts1, err := fullNode1.ChainGetTipSetByHeight(ctx, max, types.EmptyTSK)
+	require.NoError(t, err)
+
+	nv22Supply, err := fullNode0.StateVMCirculatingSupplyInternal(ctx, ts0.Key())
+	require.NoError(t, err, "Failed to fetch nv22 circulating supply")
+	nv23Supply, err := fullNode1.StateVMCirculatingSupplyInternal(ctx, ts1.Key())
+	require.NoError(t, err, "Failed to fetch nv23 circulating supply")
+
+	// Unfortunately there's still some non-determinism in supply dynamics so check for equality within a tolerance
```
Comment (ZenGround0) [inline itests/supply_test.go:144]: This is a huge pain and I'm not sure where the non-determinism is coming from -- we are measuring two identically set up test networks with a single miner at the exact same epoch.  If I had to guess I'd say sensitivity of IP function to variations in message landing time.

Though I would love to understand why it is happening and eliminate non determinism from the itest framework, for the purposes of testing FIP 0065 isolating out the non-determinism is probably a lot of work for something that is ultimately not important.

File: itests/supply_test.go [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+		Client:               client,
+		Provider:             provider,
+		ClientCollateral:     big.Zero(),
+		ProviderCollateral:   providerLocked,
+		StartEpoch:           10000,
+		EndEpoch:             10000 + abi.ChainEpoch(duration),
+		StoragePricePerEpoch: ppe,
+	}
+	buf := bytes.NewBuffer(nil)
+	proposal.MarshalCBOR(buf)
```
Comment (rvagg) [inline itests/supply_test.go:184]:
```suggestion
	require.NoError(t, proposal.MarshalCBOR(buf))
```

File: itests/supply_test.go (hunk 3 above)
Comment (rvagg) [reply itests/supply_test.go:144]: that's kind of weird, is the difference unstable across different test runs? this wouldn't be due to the f099 changes would it?

**Review by rvagg — APPROVED** [review]
👍, although the need for a difference tolerance seems a bit concerning without an explanation

File: itests/supply_test.go (hunk 3 above)
Comment (ZenGround0) [reply itests/supply_test.go:144]: Any ensemble run has the difference.  So between test runs and between the two ensembles in every test run.

Tbh it's not particularly surprising from the reputation of economics in new networks being difficult to work with.  And the sensitivity to message timing of filtering on power and circ supply[ within actors](https://github.com/filecoin-project/ref-fvm/blob/d10baf82857df234b37404507df8bbaa581990ac/shared/src/smooth/alpha_beta_filter.rs) is a hypothesis I'm reasonably ok with.

Comment (ZenGround0) [reply itests/supply_test.go:144]: The difference in locked balance is almost certainly coming from differences in total InitialPledge between runs as Burnt is kept in a different category.

**Comment by anorth** [comment]
Thanks!

**Pull request merged**